<!doctype html><html lang="en"><head><meta charset="utf-8"/><meta name="viewport" content="width=device-width,initial-scale=1"/><title>RNA-seq Viewer</title><script>const REFERENCE_DATA_ROOT = 'gs://macarthurlab-rnaseq/reference_tracks/GTEx_ref_data'
      const REFERENCE_DATA_INFO_LIST = [
        {
          name: 'GTEx 100 Muscle',
          description: '100 randomly-chosen GTEx v3 muscle samples combined by summing raw coverage values and raw splice-junction-spanning read counts across all 100 samples.',
          junctions: `${REFERENCE_DATA_ROOT}/muscle_100_GTEx_samples.junctions.bed.gz`,
          coverage: `${REFERENCE_DATA_ROOT}/muscle_100_GTEx_samples.bigWig`,
        },
        {
          name: 'GTEx 100 Blood',
          description: '100 randomly-chosen GTEx v3 blood samples combined by summing raw coverage values and raw splice-junction-spanning read counts across all 100 samples.',
          junctions: `${REFERENCE_DATA_ROOT}/blood_100_GTEx_samples.junctions.bed.gz`,
          coverage: `${REFERENCE_DATA_ROOT}/blood_100_GTEx_samples.bigWig`,
        },
        {
          name: 'GTEx 100 Fibs',
          description: '100 randomly-chosen GTEx v3 fibroblast samples combined by summing raw coverage values and raw splice-junction-spanning read counts across all 100 samples.',
          junctions: `${REFERENCE_DATA_ROOT}/fibs_100_GTEx_samples.junctions.bed.gz`,
          coverage: `${REFERENCE_DATA_ROOT}/fibs_100_GTEx_samples.bigWig`,
        },
        {
          name: 'GTEx All Muscle',
          description: 'All 803 GTEx v3 muscle samples combined by summing raw coverage values and raw splice-junction-spanning read counts across all samples.',
          junctions: `${REFERENCE_DATA_ROOT}/GTEX_muscle.803_samples.junctions.bed.gz`,
          coverage: `${REFERENCE_DATA_ROOT}/GTEX_muscle.803_samples.bigWig`,
        },
        {
          name: 'GTEx All Blood',
          description: 'All 755 GTEx v3 blood samples combined by summing raw coverage values and raw splice-junction-spanning read counts across all samples.',
          junctions: `${REFERENCE_DATA_ROOT}/GTEX_blood.755_samples.junctions.bed.gz`,
          coverage: `${REFERENCE_DATA_ROOT}/GTEX_blood.755_samples.bigWig`,
        },
        {
          name: 'GTEx All Fibs',
          description: 'All 504 GTEx v3 fibroblast samples combined by summing raw coverage values and raw splice-junction-spanning read counts across all samples.',
          junctions: `${REFERENCE_DATA_ROOT}/GTEX_fibs.504_samples.junctions.bed.gz`,
          coverage: `${REFERENCE_DATA_ROOT}/GTEX_fibs.504_samples.bigWig`,
        },
        {
          name: 'GTEx All Muscle - Norm.',
          description: 'All 803 GTEx v3 muscle samples combined by summing raw coverage values across all samples and also summing normalized splice-junction-spanning read counts across all samples. The normalization is done by computing the normalized read count for each junction as normalized_read_count = raw_read_count * scalar. Here scalar = average_unique_reads_per_muscle_sample / (total_unqiue_reads_in_this_sample * number_of_muscle_samples), and average_unique_reads_per_muscle_sample = (total_unqiue_reads_in_all_muscle_samples / number_of_muscle_samples)',
          junctions: `${REFERENCE_DATA_ROOT}/GTEX_muscle.803_samples.normalized.junctions.bed.gz`,
          coverage: `${REFERENCE_DATA_ROOT}/GTEX_muscle.803_samples.bigWig`,
        },
        {
          name: 'GTEx All Blood - Norm.',
          description: 'All 755 GTEx v3 blood samples combined by summing raw coverage values across all samples and also summing normalized splice-junction-spanning read counts across all samples. The normalization is done by computing the normalized read count for each junction as normalized_read_count = raw_read_count * scalar. Here scalar = average_unique_reads_per_blood_sample / (total_unqiue_reads_in_this_sample * number_of_blood_samples), and average_unique_reads_per_blood_sample = (total_unqiue_reads_in_all_blood_samples / number_of_blood_samples)',
          junctions: `${REFERENCE_DATA_ROOT}/GTEX_blood.755_samples.normalized.junctions.bed.gz`,
          coverage: `${REFERENCE_DATA_ROOT}/GTEX_blood.755_samples.bigWig`,
        },
        {
          name: 'GTEx All Fibs - Norm.',
          description: 'All 504 GTEx v3 fibroblast samples combined by summing raw coverage values across all samples and also summing normalized splice-junction-spanning read counts across all samples. The normalization is done by computing the normalized read count for each junction as normalized_read_count = raw_read_count * scalar. Here scalar = average_unique_reads_per_fibs_sample / (total_unqiue_reads_in_this_sample * number_of_fibs_samples), and average_unique_reads_per_fibs_sample = (total_unqiue_reads_in_all_fibs_samples / number_of_fibs_samples)',
          junctions: `${REFERENCE_DATA_ROOT}/GTEX_fibs.504_samples.normalized.junctions.bed.gz`,
          coverage: `${REFERENCE_DATA_ROOT}/GTEX_fibs.504_samples.bigWig`,
        },
        /*
        {
          name: 'Gencode v32 Genes',
          description: 'Comprehensive gene annotations for GRCh38 reference chromosomes. Source: https://www.gencodegenes.org/human/release_32.html',
          annotation: 'gs://macarthurlab-rnaseq/reference_tracks/gencode.v32.annotation.sorted.gtf.gz'
        },
        {
          name: 'RefSeq Genes',
          path: 'https://s3.amazonaws.com/igv.org.genomes/hg38/refGene.sorted.txt.gz'
        }
        */
      ]

      const SAMPLE_INFO_LIST = []

      window.INITIAL_SAMPLES_IN_CATEGORIES = [
        {
          categoryName: 'Reference Data',
          samples: REFERENCE_DATA_INFO_LIST,
        },
        {
          categoryName: 'Samples',
          samples: SAMPLE_INFO_LIST,
        },
      ]

      window.IGV_SETTINGS = {
        genome: 'hg38',
        locus: 'chr15:92,835,700-93,031,800',
        sjOptions: {
          trackHeight: 170,
          showCoverage: true,
          showJunctions: true,
          hideAnnotated: false,
          hideUnannotated: false,
          colorBy: 'strand',
          minUniquelyMappedReads: 1,
          minTotalReads: 1,
          maxFractionMultiMappedReads: 1,
          minSplicedAlignmentOverhang: 0,
          thicknessBasedOn: 'numUniqueReads', //options: numUniqueReads (default), numReads, isAnnotatedJunction
          bounceHeightBasedOn: 'random', //options: random (default), distance, thickness
          labelUniqueReadCount: true,
          labelMultiMappedReadCount: false,
          labelTotalReadCount: false,
          labelMotif: false,
          labelAnnotatedJunction: false,
          labelAnnotatedJunctionValue: " [A]",
        },
        vcfOptions: {
          displayMode: 'EXPANDED',
        },
        bamOptions: {
          trackHeight: 200,
          showSoftClips: true,
          alignmentShading: 'strand',
        },
        samplesInCategories: INITIAL_SAMPLES_IN_CATEGORIES,
        selectedSampleNamesByCategoryName: {},
      }</script><link rel="icon" href="/rnaseq-viewer/favicon.ico"/><link rel="stylesheet" href="//cdn.jsdelivr.net/npm/semantic-ui@2.4.2/dist/semantic.min.css"/><link href="/rnaseq-viewer/static/css/main.afab66cc.chunk.css" rel="stylesheet"></head><body><div id="root"></div><script src="https://apis.google.com/js/api.js"></script><script async src="https://www.googletagmanager.com/gtag/js?id=UA-153143382-1"></script><script>function gtag(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","UA-153143382-1")</script><script>!function(p){function e(e){for(var r,t,n=e[0],o=e[1],u=e[2],a=0,i=[];a<n.length;a++)t=n[a],Object.prototype.hasOwnProperty.call(f,t)&&f[t]&&i.push(f[t][0]),f[t]=0;for(r in o)Object.prototype.hasOwnProperty.call(o,r)&&(p[r]=o[r]);for(s&&s(e);i.length;)i.shift()();return c.push.apply(c,u||[]),l()}function l(){for(var e,r=0;r<c.length;r++){for(var t=c[r],n=!0,o=1;o<t.length;o++){var u=t[o];0!==f[u]&&(n=!1)}n&&(c.splice(r--,1),e=a(a.s=t[0]))}return e}var t={},f={1:0},c=[];function a(e){if(t[e])return t[e].exports;var r=t[e]={i:e,l:!1,exports:{}};return p[e].call(r.exports,r,r.exports,a),r.l=!0,r.exports}a.m=p,a.c=t,a.d=function(e,r,t){a.o(e,r)||Object.defineProperty(e,r,{enumerable:!0,get:t})},a.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},a.t=function(r,e){if(1&e&&(r=a(r)),8&e)return r;if(4&e&&"object"==typeof r&&r&&r.__esModule)return r;var t=Object.create(null);if(a.r(t),Object.defineProperty(t,"default",{enumerable:!0,value:r}),2&e&&"string"!=typeof r)for(var n in r)a.d(t,n,function(e){return r[e]}.bind(null,n));return t},a.n=function(e){var r=e&&e.__esModule?function(){return e.default}:function(){return e};return a.d(r,"a",r),r},a.o=function(e,r){return Object.prototype.hasOwnProperty.call(e,r)},a.p="/rnaseq-viewer/";var r=this.webpackJsonprnaseq_viewer_react_app=this.webpackJsonprnaseq_viewer_react_app||[],n=r.push.bind(r);r.push=e,r=r.slice();for(var o=0;o<r.length;o++)e(r[o]);var s=n;l()}([])</script><script src="/rnaseq-viewer/static/js/2.18413b46.chunk.js"></script><script src="/rnaseq-viewer/static/js/main.b68ef3ce.chunk.js"></script></body></html>