<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <title>RNA-seq Viewer</title>

    <script>
      const REFERENCE_DATA_ROOT = 'gs://macarthurlab-rnaseq/reference_tracks/GTEx_ref_data'
      const REFERENCE_DATA_INFO_LIST = [
        {
          name: 'GTEx 100 Muscle',
          description: '100 randomly-chosen GTEx v3 muscle samples combined by summing raw coverage values and raw splice-junction-spanning read counts across all 100 samples.',
          junctions: `${REFERENCE_DATA_ROOT}/muscle_100_GTEx_samples.junctions.bed.gz`,
          coverage: `${REFERENCE_DATA_ROOT}/muscle_100_GTEx_samples.bigWig`,
        },
        {
          name: 'GTEx 100 Blood',
          description: '100 randomly-chosen GTEx v3 blood samples combined by summing raw coverage values and raw splice-junction-spanning read counts across all 100 samples.',
          junctions: `${REFERENCE_DATA_ROOT}/blood_100_GTEx_samples.junctions.bed.gz`,
          coverage: `${REFERENCE_DATA_ROOT}/blood_100_GTEx_samples.bigWig`,
        },
        {
          name: 'GTEx 100 Fibs',
          description: '100 randomly-chosen GTEx v3 fibroblast samples combined by summing raw coverage values and raw splice-junction-spanning read counts across all 100 samples.',
          junctions: `${REFERENCE_DATA_ROOT}/fibs_100_GTEx_samples.junctions.bed.gz`,
          coverage: `${REFERENCE_DATA_ROOT}/fibs_100_GTEx_samples.bigWig`,
        },
        {
          name: 'GTEx All Muscle',
          description: 'All 803 GTEx v3 muscle samples combined by summing raw coverage values and raw splice-junction-spanning read counts across all samples.',
          junctions: `${REFERENCE_DATA_ROOT}/GTEX_muscle.803_samples.junctions.bed.gz`,
          coverage: `${REFERENCE_DATA_ROOT}/GTEX_muscle.803_samples.bigWig`,
        },
        {
          name: 'GTEx All Blood',
          description: 'All 755 GTEx v3 blood samples combined by summing raw coverage values and raw splice-junction-spanning read counts across all samples.',
          junctions: `${REFERENCE_DATA_ROOT}/GTEX_blood.755_samples.junctions.bed.gz`,
          coverage: `${REFERENCE_DATA_ROOT}/GTEX_blood.755_samples.bigWig`,
        },
        {
          name: 'GTEx All Fibs',
          description: 'All 504 GTEx v3 fibroblast samples combined by summing raw coverage values and raw splice-junction-spanning read counts across all samples.',
          junctions: `${REFERENCE_DATA_ROOT}/GTEX_fibs.504_samples.junctions.bed.gz`,
          coverage: `${REFERENCE_DATA_ROOT}/GTEX_fibs.504_samples.bigWig`,
        },
        {
          name: 'GTEx All Muscle - Norm.',
          description: 'All 803 GTEx v3 muscle samples combined by summing raw coverage values across all samples and also summing normalized splice-junction-spanning read counts across all samples. The normalization is done by computing the normalized read count for each junction as normalized_read_count = raw_read_count * scalar. Here scalar = average_unique_reads_per_muscle_sample / (total_unqiue_reads_in_this_sample * number_of_muscle_samples), and average_unique_reads_per_muscle_sample = (total_unqiue_reads_in_all_muscle_samples / number_of_muscle_samples)',
          junctions: `${REFERENCE_DATA_ROOT}/GTEX_muscle.803_samples.normalized.junctions.bed.gz`,
          coverage: `${REFERENCE_DATA_ROOT}/GTEX_muscle.803_samples.bigWig`,
        },
        {
          name: 'GTEx All Blood - Norm.',
          description: 'All 755 GTEx v3 blood samples combined by summing raw coverage values across all samples and also summing normalized splice-junction-spanning read counts across all samples. The normalization is done by computing the normalized read count for each junction as normalized_read_count = raw_read_count * scalar. Here scalar = average_unique_reads_per_blood_sample / (total_unqiue_reads_in_this_sample * number_of_blood_samples), and average_unique_reads_per_blood_sample = (total_unqiue_reads_in_all_blood_samples / number_of_blood_samples)',
          junctions: `${REFERENCE_DATA_ROOT}/GTEX_blood.755_samples.normalized.junctions.bed.gz`,
          coverage: `${REFERENCE_DATA_ROOT}/GTEX_blood.755_samples.bigWig`,
        },
        {
          name: 'GTEx All Fibs - Norm.',
          description: 'All 504 GTEx v3 fibroblast samples combined by summing raw coverage values across all samples and also summing normalized splice-junction-spanning read counts across all samples. The normalization is done by computing the normalized read count for each junction as normalized_read_count = raw_read_count * scalar. Here scalar = average_unique_reads_per_fibs_sample / (total_unqiue_reads_in_this_sample * number_of_fibs_samples), and average_unique_reads_per_fibs_sample = (total_unqiue_reads_in_all_fibs_samples / number_of_fibs_samples)',
          junctions: `${REFERENCE_DATA_ROOT}/GTEX_fibs.504_samples.normalized.junctions.bed.gz`,
          coverage: `${REFERENCE_DATA_ROOT}/GTEX_fibs.504_samples.bigWig`,
        },
        /*
        {
          name: 'Gencode v32 Genes',
          description: 'Comprehensive gene annotations for GRCh38 reference chromosomes. Source: https://www.gencodegenes.org/human/release_32.html',
          annotation: 'gs://macarthurlab-rnaseq/reference_tracks/gencode.v32.annotation.sorted.gtf.gz'
        },
        {
          name: 'RefSeq Genes',
          path: 'https://s3.amazonaws.com/igv.org.genomes/hg38/refGene.sorted.txt.gz'
        }
        */
      ]

      const SAMPLE_INFO_LIST = []

      window.INITIAL_SAMPLES_IN_CATEGORIES = [
        {
          categoryName: 'Reference Data',
          samples: REFERENCE_DATA_INFO_LIST,
        },
        {
          categoryName: 'Samples',
          samples: SAMPLE_INFO_LIST,
        },
      ]

      window.IGV_SETTINGS = {
        genome: 'hg38',
        locus: 'chr15:92,835,700-93,031,800',
        sjOptions: {
          trackHeight: 170,
          showCoverage: true,
          showJunctions: true,
          hideAnnotated: false,
          hideUnannotated: false,
          colorBy: 'strand',
          minUniquelyMappedReads: 1,
          minTotalReads: 1,
          maxFractionMultiMappedReads: 1,
          minSplicedAlignmentOverhang: 0,
          thicknessBasedOn: 'numUniqueReads', //options: numUniqueReads (default), numReads, isAnnotatedJunction
          bounceHeightBasedOn: 'random', //options: random (default), distance, thickness
          labelUniqueReadCount: true,
          labelMultiMappedReadCount: false,
          labelTotalReadCount: false,
          labelMotif: false,
          labelAnnotatedJunction: false,
          labelAnnotatedJunctionValue: " [A]",
        },
        vcfOptions: {
          displayMode: 'EXPANDED',
        },
        bamOptions: {
          trackHeight: 200,
          showSoftClips: true,
          alignmentShading: 'strand',
        },
        samplesInCategories: INITIAL_SAMPLES_IN_CATEGORIES,
        selectedSampleNamesByCategoryName: {},
      }

    </script>
    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
    <link rel="stylesheet" href="//cdn.jsdelivr.net/npm/semantic-ui@2.4.2/dist/semantic.min.css" />
  </head>
  <body>
    <div id="root"></div>

    <script src="https://apis.google.com/js/api.js"></script>
    <script async src="https://www.googletagmanager.com/gtag/js?id=UA-153143382-1"></script>
    <script> window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date()); gtag('config', 'UA-153143382-1');
    </script>
  </body>
</html>
